### DISCUSSION TITLE
Will this work with Sonos Connect?

### DISCUSSION THREAD

**Discussion by jmorris644** [hklages/node-red-contrib-sonos-plus#250] [Q&A] [answered]
I have a Sonos connect. I have not been able to get this to work with it. I have tried the search as well as entering the ip address.

**Comment by hklages** [comment]
I don't have a connect. 

In your browser enter the following line but use your IP address:
http://your-ip:1400/xml/device_description.xml

What is the result?

**Reply by jmorris644** [reply]
`This XML file does not appear to have any style information associated with it. The document tree is shown below.
<root xmlns="urn:schemas-upnp-org:device-1-0">
<specVersion>
<major>1</major>
<minor>0</minor>
</specVersion>
<device>
<deviceType>urn:schemas-upnp-org:device:ZonePlayer:1</deviceType>
<friendlyName>192.168.1.194 - Sonos Connect</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>ZP90</modelNumber>
<modelDescription>Sonos Connect</modelDescription>
<modelName>Sonos Connect</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/ZP90</modelURL>
<softwareVersion>57.14-37250</softwareVersion>
<swGen>1</swGen>
<hardwareVersion>1.16.5.5-1.0</hardwareVersion>
<serialNum>B8-E9-37-92-91-04:A</serialNum>
<MACAddress>B8:E9:37:92:91:04</MACAddress>
<UDN>uuid:RINCON_B8E93792910401400</UDN>
<iconList>
<icon>
<id>0</id>
<mimetype>image/png</mimetype>
<width>48</width>
<height>48</height>
<depth>24</depth>
<url>/img/icon-ZP90.png</url>
</icon>
</iconList>
<minCompatibleVersion>56.0-00000</minCompatibleVersion>
<legacyCompatibleVersion>36.0-00000</legacyCompatibleVersion>
<apiVersion>1.18.9</apiVersion>
<minApiVersion>1.1.0</minApiVersion>
<displayVersion>11.7.1</displayVersion>
<extraVersion>OTP: 1.1.1(1-16-5-zp90-0.8)</extraVersion>
<roomName>Living Room</roomName>
<displayName>Connect</displayName>
<zoneType>1</zoneType>
<feature1>0x00310001</feature1>
<feature2>0x00006172</feature2>
<feature3>0x00031021</feature3>
<seriesid>P100</seriesid>
<variant>0</variant>
<internalSpeakerSize>-1</internalSpeakerSize>
<bassExtension>0.000</bassExtension>
<satGainOffset>0.000</satGainOffset>
<memory>32</memory>
<flash>32</flash>
#DEACTIVATION_STATE_TAG_AND_VALUE# #DEACTIVATION_TTL_TAG_AND_VALUE# #DEACTIVATION_DATE_TIME_TAG_AND_VALUE#
<flashRepartitioned>1</flashRepartitioned>
<ampOnTime>425</ampOnTime>
<retailMode>0</retailMode>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:AlarmClock:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AlarmClock</serviceId>
<controlURL>/AlarmClock/Control</controlURL>
<eventSubURL>/AlarmClock/Event</eventSubURL>
<SCPDURL>/xml/AlarmClock1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:MusicServices:1</serviceType>
<serviceId>urn:upnp-org:serviceId:MusicServices</serviceId>
<controlURL>/MusicServices/Control</controlURL>
<eventSubURL>/MusicServices/Event</eventSubURL>
<SCPDURL>/xml/MusicServices1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:AudioIn:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AudioIn</serviceId>
<controlURL>/AudioIn/Control</controlURL>
<eventSubURL>/AudioIn/Event</eventSubURL>
<SCPDURL>/xml/AudioIn1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:DeviceProperties:1</serviceType>
<serviceId>urn:upnp-org:serviceId:DeviceProperties</serviceId>
<controlURL>/DeviceProperties/Control</controlURL>
<eventSubURL>/DeviceProperties/Event</eventSubURL>
<SCPDURL>/xml/DeviceProperties1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:SystemProperties:1</serviceType>
<serviceId>urn:upnp-org:serviceId:SystemProperties</serviceId>
<controlURL>/SystemProperties/Control</controlURL>
<eventSubURL>/SystemProperties/Event</eventSubURL>
<SCPDURL>/xml/SystemProperties1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ZoneGroupTopology:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ZoneGroupTopology</serviceId>
<controlURL>/ZoneGroupTopology/Control</controlURL>
<eventSubURL>/ZoneGroupTopology/Event</eventSubURL>
<SCPDURL>/xml/ZoneGroupTopology1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:GroupManagement:1</serviceType>
<serviceId>urn:upnp-org:serviceId:GroupManagement</serviceId>
<controlURL>/GroupManagement/Control</controlURL>
<eventSubURL>/GroupManagement/Event</eventSubURL>
<SCPDURL>/xml/GroupManagement1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-tencent-com:service:QPlay:1</serviceType>
<serviceId>urn:tencent-com:serviceId:QPlay</serviceId>
<controlURL>/QPlay/Control</controlURL>
<eventSubURL>/QPlay/Event</eventSubURL>
<SCPDURL>/xml/QPlay1.xml</SCPDURL>
</service>
</serviceList>
<deviceList>
<device>
<deviceType>urn:schemas-upnp-org:device:MediaServer:1</deviceType>
<friendlyName>192.168.1.194 - Sonos Connect Media Server</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>ZP90</modelNumber>
<modelDescription>Sonos Connect Media Server</modelDescription>
<modelName>Sonos Connect</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/ZP90</modelURL>
<UDN>uuid:RINCON_B8E93792910401400_MS</UDN>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:ContentDirectory:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ContentDirectory</serviceId>
<controlURL>/MediaServer/ContentDirectory/Control</controlURL>
<eventSubURL>/MediaServer/ContentDirectory/Event</eventSubURL>
<SCPDURL>/xml/ContentDirectory1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
<controlURL>/MediaServer/ConnectionManager/Control</controlURL>
<eventSubURL>/MediaServer/ConnectionManager/Event</eventSubURL>
<SCPDURL>/xml/ConnectionManager1.xml</SCPDURL>
</service>
</serviceList>
</device>
<device>
<deviceType>urn:schemas-upnp-org:device:MediaRenderer:1</deviceType>
<friendlyName>Living Room - Sonos Connect Media Renderer</friendlyName>
<manufacturer>Sonos, Inc.</manufacturer>
<manufacturerURL>http://www.sonos.com</manufacturerURL>
<modelNumber>ZP90</modelNumber>
<modelDescription>Sonos Connect Media Renderer</modelDescription>
<modelName>Sonos Connect</modelName>
<modelURL>http://www.sonos.com/products/zoneplayers/ZP90</modelURL>
<UDN>uuid:RINCON_B8E93792910401400_MR</UDN>
<serviceList>
<service>
<serviceType>urn:schemas-upnp-org:service:RenderingControl:1</serviceType>
<serviceId>urn:upnp-org:serviceId:RenderingControl</serviceId>
<controlURL>/MediaRenderer/RenderingControl/Control</controlURL>
<eventSubURL>/MediaRenderer/RenderingControl/Event</eventSubURL>
<SCPDURL>/xml/RenderingControl1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:ConnectionManager:1</serviceType>
<serviceId>urn:upnp-org:serviceId:ConnectionManager</serviceId>
<controlURL>/MediaRenderer/ConnectionManager/Control</controlURL>
<eventSubURL>/MediaRenderer/ConnectionManager/Event</eventSubURL>
<SCPDURL>/xml/ConnectionManager1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:AVTransport:1</serviceType>
<serviceId>urn:upnp-org:serviceId:AVTransport</serviceId>
<controlURL>/MediaRenderer/AVTransport/Control</controlURL>
<eventSubURL>/MediaRenderer/AVTransport/Event</eventSubURL>
<SCPDURL>/xml/AVTransport1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-sonos-com:service:Queue:1</serviceType>
<serviceId>urn:sonos-com:serviceId:Queue</serviceId>
<controlURL>/MediaRenderer/Queue/Control</controlURL>
<eventSubURL>/MediaRenderer/Queue/Event</eventSubURL>
<SCPDURL>/xml/Queue1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:GroupRenderingControl:1</serviceType>
<serviceId>urn:upnp-org:serviceId:GroupRenderingControl</serviceId>
<controlURL>/MediaRenderer/GroupRenderingControl/Control</controlURL>
<eventSubURL>/MediaRenderer/GroupRenderingControl/Event</eventSubURL>
<SCPDURL>/xml/GroupRenderingControl1.xml</SCPDURL>
</service>
<service>
<serviceType>urn:schemas-upnp-org:service:VirtualLineIn:1</serviceType>
<serviceId>urn:upnp-org:serviceId:VirtualLineIn</serviceId>
<controlURL>/MediaRenderer/VirtualLineIn/Control</controlURL>
<eventSubURL>/MediaRenderer/VirtualLineIn/Event</eventSubURL>
<SCPDURL>/xml/VirtualLineIn1.xml</SCPDURL>
</service>
</serviceList>
<X_Rhapsody-Extension xmlns="http://www.real.com/rhapsody/xmlns/upnp-1-0">
<deviceID>urn:rhapsody-real-com:device-id-1-0:sonos_1:RINCON_B8E93792910401400</deviceID>
<deviceCapabilities>
<interactionPattern type="real-rhapsody-upnp-1-0"/>
</deviceCapabilities>
</X_Rhapsody-Extension>
<qq:X_QPlay_SoftwareCapability xmlns:qq="http://www.tencent.com">QPlay:2</qq:X_QPlay_SoftwareCapability>
<iconList>
<icon>
<mimetype>image/png</mimetype>
<width>48</width>
<height>48</height>
<depth>24</depth>
<url>/img/icon-ZP90.png</url>
</icon>
</iconList>
</device>
</deviceList>
</device>
</root>`

**Comment by hklages** [comment, marked as answer]
So the SONOS device  is reachable and provides the expected answer. What happens if you create a universal node with topic household.get.groups and a config node with your IP address as above and click deploy. Does it show an error message?

**Comment by jmorris644** [comment]
So I received a valid response. I then changed the node back to player.set.mutestate and it started working. I do not know why  but I will keep an eye on it. Thanks

**Comment by hklages** [comment]
Thanks for your feedback. Let me know if you have any issues using this node.

**Comment by jmorris644** [comment]
It would be nice if the player.set.volume would take a relative number like +5.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by hklages** [comment]
... player.adjust.volume :-)

**Comment by jmorris644** [comment]
Oh man. Haha. Sorry.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jmorris644** [comment]
With this node I am using the mute and unmute to stop and start the player. I noticed two differences between using the node in this fashion and using the app to stop and start the player. 1) The mute LED stays lit and 2) the player times out being in play mode and I have to hit the play button in the app after unmuting using the node.

I looked for a play and stop command. But did not see any.

**Reply by jmorris644** [reply]
Also, I noticed those Get Outlook lines. I am not adding those personally, any idea where they might be coming from?

**Reply by hklages** [reply]
In your email client either under closing or signature.

**Comment by hklages** [comment]
My node treats single player also as group (of one player). So you can use all group command such as group.play, group.stop, ... All are documented in the Wiki.

**Reply by jmorris644** [reply]
Thanks. Interesting... I was not looking at the group commands at all as I only have the single player. But now, reading the info about groups, I understand the logic. It was just not intuitive to me.

**Comment by hklages** [comment]
... it is a bit strange on first sight but most of the users have many
players and use the group functionality and then it is very convenient.
